newIDE/app/src/Debugger/index.js

```diff
@@ -124,15 +127,34 @@ export default class Debugger extends React.Component<Props, State> {
       },
       onConnectionClosed: ({ id, debuggerIds }) => {
         this.setState(
-          ({ selectedId }) => ({
-            debuggerIds,
-            selectedId:
-              selectedId !== id
-                ? selectedId
-                : debuggerIds.length
-                ? debuggerIds[debuggerIds.length - 1]
-                : selectedId,
-          }),
+          ({
+            selectedId,
+            logs,
+            debuggerGameData,
+            profilerOutputs,
+            profilingInProgress,
+          }) => {
+            // Remove any data bound to the instance that might have been stored
+            // Otherwise it would be a data leak
```

Comment:
nitpicking:
```suggestion
            // Remove any data bound to the instance that might have been stored.
            // Otherwise this would be a memory leak.
```